apps/api/src/app/webhooks/usecases/email-webhook/email-webhook.usecase.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository, MessageRepository } from '@novu/dal';
+import { ChannelTypeEnum, EmailProviderIdEnum } from '@novu/shared';
+import { IEmailEventBody, IEmailProvider } from '@novu/stateless';
+import { MailFactory } from '../../../events/services/mail-service/mail.factory';
+import { decryptCredentials } from '../../../shared/services/encryption';
+import { EmailWebhookCommand } from './email-webhook.command';
+
+interface IWebhookResult {
+  id: string;
+  event: IEmailEventBody;
+}
+
+@Injectable()
+export class EmailWebhook {
+  private mailFactory = new MailFactory();
+  private provider: IEmailProvider;
+
+  constructor(private integrationRepository: IntegrationRepository, private messageRepository: MessageRepository) {}
+
+  async execute(command: EmailWebhookCommand): Promise<IWebhookResult[]> {
+    const providerId = command.providerId as EmailProviderIdEnum;
+    const body = command.body;
+
+    const integration: IntegrationEntity = await this.integrationRepository.findOne({
```

Comment:
Are not using Integration module here anymore, because if we like to make webhooks module its own service it needs to be easily detached from this service.